lib/krikri/enrichments/iso_enrich.rb

```diff
@@ -0,0 +1,150 @@
+  #   lang.providedLabel = 'fin', 'eng'
+  #   iso.enrich_value(lang)     # matches ['http://lexvo.org/id/iso639-3/fin',
+  #                              #          'http://lexvo.org/id/iso639-3/eng']
+  #
+  # Label matches are cached within the enrichment instance.
+  #
+  # @see DPLA::MAP::Controlled::Language
+  # @see http://www.lexvo.org/
+  class IsoEnrich
+    include Krikri::FieldEnrichment
+
+    TERMS = DPLA::MAP::Controlled::Language.list_terms.freeze
+    QNAMES = TERMS.map { |t| t.qname[1] }.freeze
+
+    ##
+    # Runs the enrichement against a node. Can match literal values, and Language
+    # values with a provided label.
+    #
+    # @example with a matching value
+    #   lang = enrich_value('finnish')
+    #   #=> #<DPLA::MAP::Controlled::Language:0x3f(default)>
+    #   lang.providedLabel
+    #   #=> ['finnish']
+    #   lang.exactMatch.map(&:to_term)
+    #   #=> [#<RDF::Vocabulary::Term:0x9b URI:http://lexvo.org/id/iso639-3/fin>]
+    #
+    # @example with no match
+    #   lang = enrich_value('moomin')
+    #   #=> #<DPLA::MAP::Controlled::Language:0x3f(default)>
+    #   lang.providedLabel
+    #   #=> ['moomin']
+    #   lang.exactMatch
+    #   #=> []
+    #
+    # @param value [ActiveTriples::Resource, #to_s]
+    # @return [DPLA::MAP::Controlled::Language, nil] a resource representing the
+    #   language match.
+    def enrich_value(value)
+      return enrich_node(value) if value.is_a?(ActiveTriples::Resource) &&
+        value.node?
+      return value if value.is_a?(DPLA::MAP::Controlled::Language)
+      return nil if value.is_a?(ActiveTriples::Resource)
+      enrich_literal(value)
+    end
+
+    ##
+    # Runs the enrichment over a specific node, accepting an
+    # `ActiveTriples::Resource` with a provided label and returning a new node
+    # with a lexvo match.
+    #
+    # @param value [ActiveTriples::Resource] a resource with a
+    #   `dpla:providedLabel`
+    # @return [Array<ActiveTriples::Resource>, ActiveTriples::Resource]
+    def enrich_node(value)
+      labels = value.get_values(RDF::DPLA.providedLabel)
+      return value if labels.empty?
+      labels.map { |label| enrich_literal(label) }
+    end
+
+    ##
+    # Runs the enrichment over a string.
+    #
+    # @param label [#to_s] the string to match
+    # @return [ActiveTriples::Resource] a blank node with a `dpla:providedLabel`
+    #   of `label` and a `skos:exactMatch` of the matching lexvo language,
+    #   if any
+    def enrich_literal(label)
+      return nil unless label.to_s
+      node = DPLA::MAP::Controlled::Language.new()
+      node.providedLabel = label
+      node.exactMatch = match_iso(label.to_s) || match_label(label.to_s)
+      node
+    end
+
+    ##
+    # Converts string or symbol for a three letter language code to an
+    # `ActiveTriples::Resource`.
+    #
+    # @param code [#to_sym] a three letter iso code
+    # @return [DPLA::MAP::Controlled::Language]
+    def match_iso(code)
```

Comment:
Yes, I believe that's deliberately left out of scope, for now. I'd propose to reintroduce it if/when it comes up.  It might be worth softening the language on the class-level documentation slightly to allow that this could/would be the place to convert other language formats to Lexvo.